### PR TITLE
Improve Button docs and examples

### PR DESCRIPTION
**Pull request by jhoerr**

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -54,24 +53,28 @@ const buttonModifierAndStyle = (variant?: VariantType, modifier?: ModifierType)
...
-const buttonSize = (size?: SizeType) => (size !== undefined ? `${buttonClass}--${size}` : None);
+const buttonSize = (props: ButtonProps) => 
+  props.size
+  ? `${buttonClass}--${props.size}` 
+  : '';
 
 export const Button: React.SFC<ButtonProps & React.ButtonHTMLAttributes<HTMLButtonElement>> = 
-  ({ className, children, id = Rivet.shortuid(), innerRef, modifier, onClick, size, variant, ...attrs}) => {
-  const classes = classNames(buttonModifierAndStyle(variant, modifier), buttonSize(size), className);
+  ({ className, children, id = Rivet.shortuid(), innerRef, onClick, ...props}) => {
```
Comment (athill) [inline src/components/Button/Button.tsx]: Naming things. I agree `...props` makes more sense than `...attrs`, as that's what they are called in React. My hesitation is that we sometimes use `props` for all the props passed to a component (not destructured). I actually generally use `...others` in my personal projects, but I'm not claiming that's a good name. I'm fine with `...props` over `...attrs`, but thought I'd point out the overload.

Comment (jwbennet) [reply src/components/Button/Button.tsx]: This should be changed back to `attrs` since that's the standard the rest of the project uses

Comment (jwbennet) [reply src/components/Button/Button.tsx]: I'll also add that I really prefer spreading the props that this component actually uses.  I understand the workaround you're doing here, but if I was looking at this component to figure out what it did the props line is one of the first places I look.

File: src/components/Button/SegmentedButton.md [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
+    <Button onClick={() => { console.log('Click 3!') }}>Primary Three</Button>
+</SegmentedButton>
+```
+
+### Secondary Modifier
+```jsx
+<SegmentedButton>
+    <Button modifier="secondary" onClick={() => { console.log('Click 1!') }}>Secondary One</Button>
+    <Button modifier="secondary" onClick={() => { console.log('Click 2!') }}>Secondary Two</Button>
+    <Button modifier="secondary" onClick={() => { console.log('Click 3!') }}>Secondary Three</Button>
```
Comment (jwbennet) [inline src/components/Button/SegmentedButton.md]: This example is fine, but nearly every other example in the project shows how to emulate the examples from the Rivet core documentation so this one does not match up with those.

File: src/components/Button/Button.tsx (hunk 1 above)
Comment (jhoerr) [reply src/components/Button/Button.tsx]: Thanks for the feedback -- I'll take another swing.

**Pull request merged**